app/models/services/crossref/resource.rb

```diff
@@ -0,0 +1,52 @@
+class Crossref
+  class Resource
+    attr_accessor :id, :response
+
+    def initialize id
+      self.id = id.to_s
+      endpoint = URI.parse("http://api.crossref.org/works/#{self.id}")
```

Comment:
also wondering if it would be worthwhile to split out this html out with the html you're using for the crossref autofill similar to what you did for pubmed.  Now that I'm thinking outloud, maybe not because you cant grab the html for javascript unless you use ujs.